### PR TITLE
Add pool search to home page.

### PR DESCRIPTION
**Pull request by frankie13**
Hmm, I wrote methods in this one that are instantly deprecated.  We should probably discuss that!

### CODE REVIEW

File: spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/DashboardHelperService.java [hunk 1]
```diff
@@ -176,6 +177,41 @@ public JSONObject searchProject(HttpSession session, JSONObject json) {
...
+        pools = new ArrayList<Pool>(requestManager.listAllPoolsWithLimit(50));
+      }
+
+      if (pools.size() > 0) {
+        Collections.sort(pools);
+        Collections.reverse(pools);
+        for (Pool p : pools) {
+          // yuck-o
+          b.append("<a class=\"dashboardresult\" href=\"/miso/pool/" + p.getId()
+              + "\"><div onMouseOver=\"this.className=&#39dashboardhighlight&#39\" onMouseOut=\"this.className=&#39dashboard&#39\" class=\"dashboard\">");
```
Comment (apmasell) [inline spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/DashboardHelperService.java]: I'm very confused about `&#39`. Should there be a `;` at the end? Also, doesn't `JSONObjectResponse` escape the string? If so, doesn't that mean `'` is legal?

File: spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/DashboardHelperService.java [hunk 2]
```diff
@@ -176,6 +177,41 @@ public JSONObject searchProject(HttpSession session, JSONObject json) {
...
+        pools = new ArrayList<Pool>(requestManager.listAllPoolsBySearch(searchStr));
+      } else {
+        pools = new ArrayList<Pool>(requestManager.listAllPoolsWithLimit(50));
+      }
+
+      if (pools.size() > 0) {
+        Collections.sort(pools);
+        Collections.reverse(pools);
+        for (Pool p : pools) {
+          // yuck-o
```
Comment (callunity) [inline spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/DashboardHelperService.java]: If you're looking for something slightly less yucky, @apmasell did something nice [here](https://github.com/TGAC/miso-lims/blob/develop/miso-web/src/main/webapp/scripts/pool_ajax.js#L960).

File: sqlstore/src/test/java/uk/ac/bbsrc/tgac/miso/sqlstore/SQLPoolDAOTest.java [hunk 3]
```diff
@@ -284,6 +284,31 @@ public void testListReadyByPlatformAndSearchWithNullQuery() throws IOException {
...
+  @SuppressWarnings("deprecation")
+  @Test
+  public void testListBySearchWithNullQuery() throws IOException {
+    assertEquals(0, dao.listBySearch(null).size());
+  }
+
+  @SuppressWarnings("deprecation")
+  @Test
+  public void testListBySearchWithBadQuery() throws IOException {
+    assertEquals(0, dao.listBySearch(";DROP TABLE Users;").size());
```
Comment (apmasell) [inline sqlstore/src/test/java/uk/ac/bbsrc/tgac/miso/sqlstore/SQLPoolDAOTest.java]: https://xkcd.com/327/

**Comment by apmasell** [comment]
:+1:

**Comment by callunity** [comment]
:+1:

File: spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/DashboardHelperService.java (hunk 1 above)
Comment (frankie13) [inline spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/DashboardHelperService.java]: ...yeah.  This whole approach needs review.  What I'll do is raise a ticket to move front end code out of javaland :facepunch:

**Pull request merged**